core/templates/dev/head/editor/ExplorationPreview.js

```diff
@@ -18,19 +18,29 @@
  */
 
 oppia.controller('ExplorationPreview', [
-  '$scope', '$timeout', 'LearnerParamsService', 'explorationData',
+  '$scope', '$timeout', 'LearnerParamsService',
+  'explorationData', 'editorContextService',
   'explorationStatesService', 'explorationInitStateNameService',
   'explorationParamSpecsService', 'explorationTitleService',
   'explorationCategoryService', 'explorationParamChangesService',
   'explorationGadgetsService', 'oppiaPlayerService',
   function(
-      $scope, $timeout, LearnerParamsService, explorationData,
+      $scope, $timeout, LearnerParamsService,
+      explorationData, editorContextService,
       explorationStatesService, explorationInitStateNameService,
       explorationParamSpecsService, explorationTitleService,
       explorationCategoryService, explorationParamChangesService,
       explorationGadgetsService, oppiaPlayerService) {
     $scope.isExplorationPopulated = false;
-    explorationData.getData().then(function() {
+    explorationData.getData(ignoreCache = true).then(function() {
```

Comment:
Would you mind helping me understand, in more detail, why ignoreCache is needed? I confess that I don't really get it...

I.e., when you say "the state wouldn't update properly", what exactly do you mean? The main change we're making here is which initial state the preview is starting from. I don't see why this has any bearing on the exploration data being retrieved.